### PR TITLE
Add section on migration planning openldap -> 389 DS

### PR DESCRIPTION
**Pull request by cjschroder**
re: re  Jira SLE-14732

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [ ] SLE 15 SP4/openSUSE Leap 15.4 *(if applicable to 15 SP4 _only_ -- do not merge yet!)*
  - [x] SLE 15 SP3/openSUSE Leap 15.3 *(current `main`, no backport necessary)*
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
  - [ ] SLE 15 GA
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4
  - [ ] SLE 12 SP3

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Comment by cjschroder** [comment]
@Firstyear For this PR I poached liberally from https://directory.fedoraproject.org/docs/389ds/howto/howto-openldap-migration.html. I have one question: in the table comparing features between OpenLDAP and 389 DS, what does "SLE/SUSE/2.0.0 only due to Rust requirement" mean? What are "SLE/SUSE/2.0.0"? Would it be better to specify the Rust requirements?

**Comment by Firstyear** [comment]
> @Firstyear For this PR I poached liberally from https://directory.fedoraproject.org/docs/389ds/howto/howto-openldap-migration.html. I have one question: in the table comparing features between OpenLDAP and 389 DS, what does "SLE/SUSE/2.0.0 only due to Rust requirement" mean? What are "SLE/SUSE/2.0.0"? Would it be better to specify the Rust requirements?

That's a note for developers, not for us. For our purposes, it's working for us at suse :)

File: xml/security_ldap_migrate_test.xml [hunk 1]
```diff
@@ -179,4 +179,150 @@ You should now review your instance configuration and data:
...
+      <entry>Database Import</entry>
+      <entry>LDIF</entry>
+      <entry>LDIF</entry>
+      <entry>Yes, supported by migration tool</entry>
+     </row>
+     <row>
+      <entry>Password hashes</entry>
+      <entry>Varies</entry>
+      <entry>Varies</entry>
+      <entry>Yes, all formats supported excluding ARGON2</entry>
```
Comment (Firstyear) [inline xml/security_ldap_migrate_test.xml:298]: ARGON2 (from openldap2-contrib) 

I think is the package it's in .... I'll double check to be sure.

File: xml/security_ldap_migrate_test.xml [hunk 2]
```diff
@@ -179,4 +179,150 @@ You should now review your instance configuration and data:
...
+     </row>
+     <row>
+      <entry>oldap 2 ds repl</entry>
```
Comment (Firstyear) [inline xml/security_ldap_migrate_test.xml:301]: Probably good to expand these to "OpenLDAP to 389-ds Replication"

File: xml/security_ldap_migrate_test.xml [hunk 3]
```diff
@@ -179,4 +179,150 @@ You should now review your instance configuration and data:
...
+      <entry>-</entry>
+      <entry>-</entry>
+      <entry>No, no mechanism to replicate to 389 is possible</entry>
+     </row>
+     <row>
+      <entry>ds 2 oldap repl</entry>
```
Comment (Firstyear) [inline xml/security_ldap_migrate_test.xml:307]: Please do NOT list this in the docs. We actually don't want CU using this, there is only one CU we want to support doing it, and we want them to ask us.

File: xml/security_ldap_migrate_test.xml [hunk 4]
```diff
@@ -179,4 +179,150 @@ You should now review your instance configuration and data:
...
+      <entry>-</entry>
+      <entry>SyncREPL</entry>
+      <entry>Yes (SLE/SUSE/2.0.0 only due to Rust requirement, limitations exist)</entry>
+     </row>
+     <row>
+      <entry>Time-based one-time password(TOTP)</entry>
+      <entry>TOTP overlap</entry>
```
Comment (Firstyear) [inline xml/security_ldap_migrate_test.xml:314]: overlay (not overlap).

File: xml/security_ldap_migrate_test.xml [hunk 5]
```diff
@@ -179,4 +179,150 @@ You should now review your instance configuration and data:
...
+      <entry>-</entry>
+      <entry>No, may be supported in the future</entry>
+     </row>
+     <row>
+      <entry>EntryUUID</entry>
+      <entry>Part of OpenLDAP</entry>
+      <entry>Plugin</entry>
+      <entry>Yes (SLE/SUSE/2.0.0 only due to Rust requirement)</entry>
```
Comment (Firstyear) [inline xml/security_ldap_migrate_test.xml:322]: Just say Yes here.

**Review by Firstyear — CHANGES_REQUESTED** [review]

File: xml/security_ldap_migrate_test.xml (hunk 1 above)
Comment (cjschroder) [reply xml/security_ldap_migrate_test.xml:298]: SLES has the libargon2-1 package. openSUSE has argon2, python2-argon2-cffi, libargon2-1, and some others.

Comment (Firstyear) [reply xml/security_ldap_migrate_test.xml]: Yes, but 389 doesn't support importing the argon2 hashs that openldap2-contrib can create :)

Comment (Firstyear) [reply xml/security_ldap_migrate_test.xml]: Maybe I should be clearer. Some password storage mechanisms in openldap aren't "built in" they are in an unsupported contrib module. That includes PBKDF2, SHA2, ARGON2 etc. So openldap2-contrib will likely use openssl or libargon2 or whatever, but for openldap2 to store the ARGON2 hash, it needs the openldap2contrib module.

And if someone is using this singular ARGON2 hashtype from openldap2-contrib, 389-ds can't support importing that one alone.

**Pull request merged**